mongo/crud_examples_test.go

```diff
@@ -0,0 +1,525 @@
+		if err == mongo.ErrNoDocuments {
+			return
+		}
+		log.Fatal(err)
+	}
+	fmt.Printf("updated document %v", updatedDocument)
+}
+
+func ExampleCollection_InsertMany() {
+	var coll *mongo.Collection
+
+	// insert documents {name: "Alice"} and {name: "Bob"}
+	// set the Ordered option to false to allow both operations to happen even if one of them errors
+	docs := []interface{}{
+		bson.D{{"name", "Alice"}},
+		bson.D{{"name", "Bob"}},
+	}
+	opts := options.InsertMany().SetOrdered(false)
+	res, err := coll.InsertMany(context.TODO(), docs, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("inserted documents with IDs %v\n", res.InsertedIDs)
+}
+
+func ExampleCollection_InsertOne() {
+	var coll *mongo.Collection
+
+	// insert the document {name: "Alice"}
+	// set the BypassDocumentValidation option to opt-out of document-level validation
+	opts := options.InsertOne().SetBypassDocumentValidation(true)
+	res, err := coll.InsertOne(context.TODO(), bson.D{{"name", "Alice"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("inserted document with ID %v\n", res.InsertedID)
+}
+
+func ExampleCollection_ReplaceOne() {
+	var coll *mongo.Collection
+
+	// replace at most one document in which the "name" field is "Bob" with {name: "Alice"}
+	// specify the Upsert option to insert {name: "Alice"} if a document matching the filter isn't found
+	opts := options.Replace().SetUpsert(true)
+	filter := bson.D{{"name", "Bob"}}
+	replacement := bson.D{{"name", "Alice"}}
+	result, err := coll.ReplaceOne(context.TODO(), filter, replacement, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if result.MatchedCount != 0 {
+		fmt.Println("matched and replaced an existing document")
+		return
+	}
+	if result.UpsertedCount != 0 {
+		fmt.Printf("inserted a new document with ID %v\n", result.UpsertedID)
+	}
+}
+
+func ExampleCollection_UpdateMany() {
+	var coll *mongo.Collection
+
+	// find all documents in which the "name" field is "Bob" and set the name to "Alice"
+	// specify the Upsert option to insert {name: "Alice"} if a document matching the filter isn't found
+	opts := options.Update().SetUpsert(true)
+	filter := bson.D{{"name", "Bob"}}
+	update := bson.D{{"$set", bson.D{{"name", "Alice"}}}}
+
+	result, err := coll.UpdateMany(context.TODO(), filter, update, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if result.MatchedCount != 0 {
+		fmt.Println("matched and replaced an existing document")
+		return
+	}
+	if result.UpsertedCount != 0 {
+		fmt.Printf("inserted a new document with ID %v\n", result.UpsertedID)
+	}
+}
+
+func ExampleCollection_UpdateOne() {
+	var coll *mongo.Collection
+
+	// find at most one document in which the "name" field is "Bob" and set the name to "Alice"
+	// specify the Upsert option to insert {name: "Alice"} if a document matching the filter isn't found
+	opts := options.Update().SetUpsert(true)
+	filter := bson.D{{"name", "Bob"}}
+	update := bson.D{{"$set", bson.D{{"name", "Alice"}}}}
```

Comment:
Again, such a weird example.

Reply:
agreed. changed to update a user's email instead.